homeassistant/components/splunk/__init__.py

```diff
@@ -75,13 +59,12 @@ def setup(hass, config):
     name = conf.get(CONF_NAME)
     entity_filter = conf[CONF_FILTER]
 
-    if use_ssl:
-        uri_scheme = "https://"
-    else:
-        uri_scheme = "http://"
+    event_collector = http_event_collector(
+        token, host, FORMAT_JSON, name, port, use_ssl
+    )
+    event_collector.SSL_verify = verify_ssl
 
-    event_collector = f"{uri_scheme}{host}:{port}/services/collector/event"
-    headers = {AUTHORIZATION: f"Splunk {token}"}
+    event_collector.check_connectivity()
```

Comment:
What happens if the connectivity check fails?

Reply:
It raises warnings for status codes 4xx, and errors for status codes 5xx. See https://github.com/georgestarcher/Splunk-Class-httpevent/blob/master/splunk_http_event_collector.py#L159

Running the connectivity check gives the users a prompt log that their configuration in Home Assistant or Splunk may be incorrect, but doesnt prevent the integration from working should Splunk be starting up or restarting.

Reply:
We want to check for bad credentials and if so fail the setup. If it's the connection that is bad we can keep trying. Can we differentiate between these errors?

Reply:
As best I can tell the library does not expose this information other than in what it logs. It would have been easy enough to do when using requests directly as its a status code of 401. The logging does clearly state the token is invalid, and as such the user could go into Splunk and change the token there or adjust their configuration.yaml. How would you like me to proceed here?

Once this PR is through, my intention is to convert this integration to using config flow, and which point I can perform a connectivity and credential check on first config.

Reply:
We need some way of validating the token before we proceed. Also with a config flow in place, the token can become invalid for some reason after configuration, and we then want to reauthenticate the user and start a new flow.

Can we make some api call that fails if the token is invalid and catch that error?

Reply:
We could interacting with Splunk directly over requests or aiohttp rather than using the library but I figured that breaks the rules. Splunk HEC tokens are static and plain text, so I don't think we need to account for them becoming invalid. It would more likely be an initial setup copy paste fail, which is currently logged with the error from Splunk.

Reply:
Maybe try using another package eg:
https://pypi.org/project/splunk-data-sender/

It seems to raise errors on failed authentication etc.

Reply:
My only concerns with that is it doesn't support batching and may require user to enable Acknowledgment in Splunk, but should be do-able.

Would I be best to submit a new PR once I have reworked this, or keep this train going?

Reply:
You can do either I think. Do what works best for you.